### PR TITLE
Draft: Fixes #11 Add allocation chart

### LINKED ISSUE #11: Make a frequency allocation chart
# Description
It would be nice to visually demonstrate the frequency allocations.

A good starting point might be to create a frequency allocation chart as this one [here](https://www.ntia.doc.gov/files/ntia/publications/january_2016_spectrum_wall_chart.pdf) for US allocations.

This could be tackled in a few steps:
## 1. Use mock data
Setup some mock data. Don't worry about a format yet or coupling your plot with how data is currently saved. Just worry about what you need for plotting.

For example, I'd suggest creating this:
```javascript
{allocations:
  [
    {band: {lowerFrequency: 90000, upperFrequency: 110000}, name: "Radio Navigation"},
    {band: {lowerFrequency: 1705000, upperFrequency: 1800000, name: "Fixed"},
    {band: {lowerFrequency: 1705000, upperFrequency: 1800000, name: "Mobile"},
    {band: {lowerFrequency: 540000000, upperFrequency: 720000000, name: "Television"}
  ]
}
```
where the frequency are in Hz. The above should probably enough.
Make sure to capture edge cases, like overlapping frequencies.

## 2. Plot the Allocations
Once you have some good mock data, use it to generate a test chart. This is really up to you. If anyone has suggestions please add them to this issue.

## 3. Connect to the actual data
Look at the latest version of this repo and check to see how the frequencies data is loaded.
This is currently changing very rapidly. If you think the data format should be changed in anyway, suggestions are welcome.

## 4. Finishing touches
We should make additional issues for finishing touches for someone else to eventually complete.
Here are some we should add:
- labels
- labels change units depending on the magnitude (ex: use kHz or MHz depending on how big the numbers are)
- hover links allow users to see descriptions
- add a color legend for types like [here](https://www.ntia.doc.gov/files/ntia/publications/january_2016_spectrum_wall_chart.pdf). this will require two issues: one to update the data with a category field if needed and one to add the legend
- anything else?

# Example
Frequency allocations should look sometihng like this (but more visually appealing suggesitons welcome)
Screenshot from 2021-12-14 16-21-58

### PR DESCRIPTION
**Pull request by ajmas**
Screenshot 2021-12-17 at 01 11 34

Still to do:

  - [ ] allow specifying a custom lookup table
  - [ ] allowing the content to scroll within container
  - [ ] toggle constrain vs scroll
  - [ ] add frequency indicators
  - [ ] rotate text if horizontal space is too small & vertical size is greater than hotizontal size

### CODE REVIEW

File: src/views/Home.vue [hunk 1]
```diff
@@ -6,6 +6,10 @@
       <router-link to="/about">About</router-link>
     </div>
 
+    <div id="nav" class="allocation-breadcrumbs">
```
Comment (jrmlhermitte) [inline src/views/Home.vue:9]: what is this segment doing or what is the intention? not too familiar with the router links (other than they allow routing within a single app)

**Review by jrmlhermitte — COMMENTED** [review]
looks neat!

can you please add a screenshot eventually for records?

File: src/components/AllocationChart.vue [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  props: {
+    allocations: {
+      type: Array,
+      default: () => [],
+      required: true
+    }
+  },
+  methods: {
+    allocationClass (/* allocation: any */) {
+      return 'alloc-type-a';
```
[the remark this answers is not in the document]
Comment (ajmas) [reply src/components/AllocationChart.vue:25]: A separate component in this situation would be overkill. Also onMount would be the wrong place.

Maybe there is a more elegant way and I’d need to look at that. This works for notes. 

 Actually makes me think we need to watch for property changes, if the region gets changed, this won’t be remounted.

File: src/views/Home.vue (hunk 1 above)
Comment (ajmas) [reply src/views/Home.vue:9]: It’s out of place for this PR but it would show the direct path of the region, from root parent to this one. Indicates all the parent regions

File: src/components/AllocationChart.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (jrmlhermitte) [reply src/components/AllocationChart.vue:25]: yeah, sounds good. to be clear, i think this is perfect as is, the viz looks great. might be good just to check if there are some standard practices for this murky intersection of styling and javascript.

good point about property changes. maybe make the second as an issue before merging this in? I am a proponent of small PR's when possible.

File: src/views/Home.vue (hunk 1 above)
Comment (jrmlhermitte) [reply src/views/Home.vue:9]: ah i see, so `ITU1 > us`  great idea. yeah, maybe also make this an issue so it's not lost?

Comment (ajmas) [reply src/views/Home.vue:9]: Will move breadcrumbs to issue #14

File: src/models.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+interface FrequencyBand {
+  lower: number;
+  upper: number;
+}
+
+interface FrequencyAllocation {
+  service: string;
+  primary: boolean;
+  footnotes?: string[];
```
Comment (jrmlhermitte) [inline src/models.ts:9]: if it's a list, does it need to be optional?
(what about just empty list)

File: src/models.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+interface FrequencyAllocationBlock {
+  band: FrequencyBand;
+  allocations: FrequencyAllocation[];
+}
+
+interface FrequencyAllocations {
+  name: string;
+  region: string;
+  parentRegion?: string;
+  year?: number;
```
Comment (jrmlhermitte) [inline src/models.ts:21]: in discord, this was not optional, but i like that this is optional better. unlikely we'll have a year to start with
